src/session_impl.cpp

```diff
@@ -5988,7 +6006,14 @@ namespace aux {
 	void session_impl::dht_announce(sha1_hash const& info_hash, int port, int flags)
 	{
 		if (!m_dht) return;
-		m_dht->announce(info_hash, port, flags, std::bind(&on_dht_get_peers, std::ref(m_alerts), info_hash, _1));
+		if (port == 0)
+			m_dht->announce(info_hash, flags
+				, std::bind(&session_impl::get_listen_port, this, transport::plaintext, _1)
+				, std::bind(&on_dht_get_peers, std::ref(m_alerts), info_hash, _1));
+		else
+			m_dht->announce(info_hash, flags
+				, [port](aux::listen_socket_handle const&) { return port; }
```

Comment:
is this case still necessary?

Reply:
it seems `port` could just be dropped instead

Reply:
I kept the `port` parameter so that a client can announce non-bittorrent services to the DHT.